gk/fib.c

```diff
@@ -108,22 +181,14 @@ neigh_get_ether_cache_locked(struct neighbor_hash_table *neigh,
 		"Failed to add a cache entry to the neighbor hash table at %s\n",
 		__func__);
 
-	return NULL;
-}
-
-/* Warning: avoid calling this function directly, prefer ether_cache_put(). */
-static int
-neigh_del_ether_cache(struct neighbor_hash_table *neigh, void *key)
-{
-	int ret = rte_hash_del_key(neigh->hash_table, key);
-	if (ret >= 0)
-		return initialize_ether_cache(&neigh->cache_tbl[ret]);
-
-	RTE_LOG(ERR, HASH,
-		"Failed to delete a cache entry to the neighbor hash table at %s - %s\n",
-		__func__, strerror(-ret));
+	if (addr->proto == ETHER_TYPE_IPv4)
+		put_arp(&addr->ip.v4, lcore_id);
+	else
+		put_nd(&addr->ip.v6, lcore_id);
```

Comment:
We need to add a `return -1;` just after this `if` statement because the LLS block will call `gk_arp_and_nd_req_cb()`, which, in turn, will call `clear_ether_cache()`. A comment explaining this just before the `return` would be appropriate.

Reply:
Done